include/hc_printf.hpp

```diff
@@ -71,8 +106,17 @@ enum PrintfPacketDataType {
 class PrintfPacket {
 public:
   void clear()             [[hc,cpu]] { type = PRINTF_UNUSED; }
-  void set(unsigned int d) [[hc,cpu]] { type = PRINTF_UNSIGNED_INT;   data.ui = d; }
-  void set(int d)          [[hc,cpu]] { type = PRINTF_SIGNED_INT;     data.i = d; }
+  void set(uint8_t d)      [[hc,cpu]] { type = PRINTF_UINT8_T;        data.uc = d; }
+  void set(int8_t d)       [[hc,cpu]] { type = PRINTF_INT8_T;         data.c = d; }
+  void set(uint16_t d)     [[hc,cpu]] { type = PRINTF_UINT16_T;       data.us = d; }
+  void set(int16_t d)      [[hc,cpu]] { type = PRINTF_INT16_T;        data.s = d; }
```

Comment:
We don't need sub-32bit int types for these since they'll be promoted to 32bit when passing to a varargs function